plugins/CustomTCMaker.cpp

```diff
@@ -107,6 +107,15 @@ CustomTCMaker::generate_opmon_data()
   info.set_tc_failed_sent_count( m_tc_failed_sent_count.load() );  
 
   this->publish(std::move(info));
+
+  if ( m_running_flag.load() && m_latency_monitoring.load() ) { 
+    opmon::TriggerLatency lat_info;
+
+    lat_info.set_latency_in( m_latency_instance.get_latency_in() );
+    lat_info.set_latency_out( m_latency_instance.get_latency_out() );
```

Comment:
I think there's a logic issue here:
`get_latency_in()` function seems to compare the object time-stamp with the current time. That's a number we want to obtain e.g. right after obtaining a new object, right after creating an object, or right before sending the object elsewhere. 

But here we're comparing it against time at opmon measurement time instead, which will add extra latency (because this opmon measurement might happen e.g. after the object was already sent, so we're comparing objects timestamp against system time after the object is already gone from the system).

So in other words, `get_current_system_time()` should be called when we take the latency measurement, not when we're sending it to opmon after the measurement.